File: Backend_REST/src/main/java/id/co/astratech/service/ObatService.java
package id.co.astratech.service;

import id.co.astratech.model.Obat;
import id.co.astratech.response.DtoResponse;

public interface ObatService {

    public DtoResponse getAllObat();

    public DtoResponse getObatById(Integer idObat);

    public DtoResponse saveObat(Obat obat);

    public DtoResponse updateStatus(Integer idObat);

    public DtoResponse updateObat(Obat obat);
}

File: Frontend/src/util/ObatService.js
import axios from "axios";

const BASE_URL = 'http://localhost:8080/obat';

export const listObat = () => axios.get(BASE_URL + '/getAllObat');
export const getObatById = (idObat) => axios.get(`${BASE_URL}/getObatById/${idObat}`);
export const saveObat = (obat) => axios.post(BASE_URL + '/saveObat',obat);
export const updateObat = (obat) => axios.post(BASE_URL + '/updateObat',obat);
export const updateStatus = (idObat) => axios.post(`${BASE_URL}/updateStatus/${idObat}`);